File: psra/service/src/main/scala/by/arhor/psra/service/CourseService.scala
package by.arhor.psra.service

import by.arhor.psra.dto.CourseDto
import by.arhor.psra.model.Course

trait CourseService extends Service[Course, CourseDto, String] {

  def addLearnersToCourse(courseId: String, learnersIds: Array[String]): CourseDto

}

File: psra/service/src/main/java/by/arhor/psra/service/CourseService.java
package by.arhor.psra.service;

import by.arhor.psra.dto.CourseDto;
import by.arhor.psra.dto.UserDto;

import java.util.List;

public interface CourseService
    extends Service<CourseDto, String>
          , CanCreate<CourseDto>
          , CanUpdate<CourseDto> {

  CourseDto addLearnersToCourse(String courseId, String[] learnersIds);

  List<UserDto> findLearnersByCourseId(String cid);

}
